### ISSUE TITLE
A class with async constructor (__new__ + __init__) reported as not awaitable

### ISSUE THREAD

**Issue by dm0** [astral-sh/ty#1254]
### Summary

The following sample code:
```python
import asyncio

class My:
    async def __new__(cls, *args, **kwargs):
        instance = super().__new__(cls)
        await instance.__init__(*args, **kwargs)

    async def __init__(self):
        print('__init__')

async def main():
    my = await My()

if __name__ == "__main__":
    asyncio.run(main())
```

reports `invalid-await` when checked with `ty check`:
```
r[invalid-await]: `My` is not awaitable
  --> obj.py:12:16
   |
11 | async def main():
12 |     my = await My()
   |                ^^^^
13 |
14 | if __name__ == "__main__":
   |
  ::: obj.py:3:7
   |
 1 | import asyncio
 2 |
 3 | class My:
   |       -- type defined here
 4 |     async def __new__(cls, *args, **kwargs):
 5 |         instance = super().__new__(cls)
   |
info: `__await__` is missing
info: rule `invalid-await` is enabled by default
``` 

### Version

0.0.1-alpha.21

**Comment by sharifhsn** [comment]
Closed by [#24357](https://github.com/astral-sh/ruff/pull/24357)

**Comment by charliermarsh** [comment]
Thank you!
